### PR TITLE
Docs: Spack Desktop Environment

### PR DESCRIPTION
**Pull request by ax3l**
Add Spack environment files for developer desktop/laptop computers.

- [x] link in the docs
- [ ] test each env
  - [x] Linux: OpenMP w/ Ascent (install + run)
  - [ ] Linux: CUDA w/ Ascent
  - [ ] Linux: ROCm w/ Ascent
  - [x] macOS M1: OpenMP w/ Ascent (install + run)
  - [ ] macOS x86_64: OpenMP w/ Ascent

## Dependencies in Spack
- [x] https://github.com/spack/spack/pull/32285
- [x] https://github.com/spack/spack/pull/32284

## Spack Version

Tested with Spack `v0.13.2-14196-g7fc78b8b0f` (tag search off, essentially post 0.18.1 `develop`)

### CODE REVIEW

File: Tools/machines/desktop/spack-macos-openmp.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  compilers:
+  # macOS
+  # preparation: you first need to install xcode (app store) and gcc (homebrew) and gpg2 (homebrew)
+  - compiler:
+      spec: apple-clang@13.1.6
+      paths:
+        cc: /usr/bin/clang
+        cxx: /usr/bin/clang++
+        f77: /opt/homebrew/bin/gfortran
+        fc: /opt/homebrew/bin/gfortran
```
Comment (ax3l) [inline Tools/machines/desktop/spack-macos-openmp.yaml:85]: We might even be able to put `null` here and compile without any Fortran compiler and thus no need to get `brew` first... test ongoing
```suggestion
        f77: null
        fc: null
```

File: Tools/machines/desktop/spack-macos-openmp.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      modules: []
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: apple-clang@13.1.6
+      paths:
+        cc: /usr/bin/clang
+        cxx: /usr/bin/clang++
+        f77: /opt/homebrew/bin/gfortran
+        fc: /opt/homebrew/bin/gfortran
```
Comment (ax3l) [inline Tools/machines/desktop/spack-macos-openmp.yaml:98]: See above:
```suggestion
        f77: null
        fc: null
```

File: Tools/machines/desktop/spack-macos-openmp.yaml (hunk 1 above)
Comment (ax3l) [reply Tools/machines/desktop/spack-macos-openmp.yaml:85]: OpenBLAS:
```
raise InstallError("OpenBLAS requires both C and Fortran compilers!")
```

**Review by RemiLehe — APPROVED** [review]

**Comment by n01r** [comment]
Hey @ax3l, on my x86_64 MacBook the [spack-macos-openmp.yaml](https://github.com/ECP-WarpX/WarpX/blob/development/Tools/machines/desktop/spack-macos-openmp.yaml) does not work like that, yet. I am missing `/opt/homebrew/bin/gfortran` in my setup and `brew cask install gfortran` as well as `brew install gfortran --cask` do not work, anymore, since the `gfortran` cask is deprecated since the end of 2020.

There is a separate repository that exists but this would install it under a different path.
https://github.com/fxcoudert/gfortran-for-macOS/releases

It is also part of `brew install gcc` now. So one could also do that and then change the path accordingly.

**Comment by ax3l** [comment]
Related follow-up: https://github.com/spack/spack/pull/32398

**Pull request merged**